elide-core/src/main/java/com/yahoo/elide/parsers/state/RecordState.java

```diff
@@ -40,7 +41,7 @@ public void handle(StateContext state, SubCollectionReadCollectionContext ctx) {
         try {
             RelationshipType type = dictionary.getRelationshipType(resource.getObject(), subCollection);
             if (type == RelationshipType.NONE) {
-                throw new InvalidCollectionException(subCollection);
+                throw new ForbiddenAccessException(subCollection);
```

Comment:
There is a missed case below in the other handle function

Reply:
Also, the other place where we throw InvalidCollectionException needs to be mapped as well.

Reply:
Addressed.